server/src/main/java/org/apache/druid/client/ImmutableDruidDataSource.java

```diff
@@ -109,6 +112,27 @@ public long getTotalSizeOfSegments()
     return totalSizeOfSegments;
   }
 
+  /**
+   * This method finds the fully overshadowed segments in this datasource
```

Comment:
What does the prefix "fully" mean in this context? Can a segment be "just" overshadowed?

Reply:
A segment can be partially overshadowed.

Reply:
The term "overshadowed" is used throughout the codebase in the *fully overshadowed* meaning. If *partially overshadowed* concept is ever used in the codebase, it should be called partiallyOvershadowed.

Compare with `determineOvershadowedSegments()` method. We should either have "FullyOveshadowed" everywhere or nowhere. I think we should have it nowhere.

Reply:
Sure, I think it makes sense to just use the word "overshadowed" here.